src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -103,12 +105,22 @@ public async Task SpecializeAsync()
             _logger.LogInformation("Starting language worker channel specialization");
             _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
             ILanguageWorkerChannel languageWorkerChannel = await GetChannelAsync(_workerRuntime);
+
             if (_workerRuntime != null && languageWorkerChannel != null)
             {
                 _logger.LogInformation("Loading environment variables for runtime: {runtime}", _workerRuntime);
-                await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
+
+                using (_metricsLogger.LatencyEvent(MetricEventNames.SpecializationSendEnvironmentReloadRequest))
+                {
+                    await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
```

Comment:
@pragnagopa -- is this is just to initialize the reload, or does this not return until it's complete?

Reply:
I think the "SpecializationEnvironmentReloadRequestResponse" (within LanguageWorkerChannel.cs) metric covers both the sending of request and getting response back. Please correct me if I'm wrong @pragnagopa

Reply:
yes, adding metrics inside LangaugeWorkerChannel will handle this. And yes, this task only completes once we receive response from the language worker